packages/babel-helpers/src/helpers.js

```diff
@@ -730,6 +733,29 @@ helpers.possibleConstructorReturn = helper("7.0.0-beta.0")`
   }
 `;
 
+helpers.createSuper = helper("7.9.0")`
+  import getPrototypeOf from "getPrototypeOf";
+  import isNativeReflectConstruct from "isNativeReflectConstruct";
+  import possibleConstructorReturn from "possibleConstructorReturn";
+
+  export default function _createSuper(Derived) {
+    return function () {
+      var Super = getPrototypeOf(Derived), result;
+      if (isNativeReflectConstruct()) {
+        for (var l = arguments.length, args = new Array(l), i = 0; i < l; i++) {
+          args[i] = arguments[i];
+        }
+        // NOTE: This doesn't work if this.__proto__.constructor has been modified.
+        var NewTarget = getPrototypeOf(this).constructor;
+        result = Reflect.construct(Super, args, NewTarget);
```

Comment:
Why we need shallow copy the `arguments`? `Reflect.construct` accepts array-like objects. https://tc39.es/ecma262/#sec-reflect.construct

Reply:
It's because directly passing `arguments` deoptimizes the function in some engines.

Reply:
What about using `Array.from`? comparing [`Reflect.construct`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Reflect/construct) with [`Array.from`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from) shows that `Array.from` is supported on every platforms with `Reflect.construct` support.

Also I am not sure if we should still write codes specifically for V8 <= 5.9 (when ignition pipeline is not enabled and when https://jsperf.com/123124235 shows 5x difference), the [usage](https://caniuse.com/usage-table) of `chrome <= 59` is less than 1% now. But since Chrome 49 is 0.52%, guess we should wait for a longer times.

Reply:
Do you mean `Array.from(arguments)`? Then it's the same as directly passing `arguments` to `Reflect.construct` (which I can do).

Reply:
Well I thought `Array.from` will shallow copy the `arguments` into a new array so to avoid the v8 <= 5.9 performance issue. But 👍 for clearer code.

Reply:
Well, passing `arguments` to `Array.from` doesn't cause a deopt anyway?